app/src/org/commcare/dalvik/activities/HomeActivityUIController.java

```diff
@@ -0,0 +1,391 @@
+package org.commcare.dalvik.activities;
+
+import android.annotation.SuppressLint;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.text.Spannable;
+import android.text.format.DateUtils;
+import android.util.Log;
+import android.util.Pair;
+import android.view.View;
+import android.view.ViewTreeObserver;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.etsy.android.grid.StaggeredGridView;
+
+import org.commcare.android.adapters.HomeScreenAdapter;
+import org.commcare.android.database.UserStorageClosedException;
+import org.commcare.android.view.SquareButtonWithNotification;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.preferences.CommCarePreferences;
+import org.commcare.suite.model.Profile;
+import org.javarosa.core.services.locale.Localization;
+
+import java.text.DateFormat;
+import java.util.Date;
+
+import in.srain.cube.views.GridViewWithHeaderAndFooter;
+
+/**
+ * Created by amstone326 on 9/17/15.
+ */
+public class HomeActivityUIController {
+
+    private final static String UNSENT_FORM_NUMBER_KEY = "unsent-number-limit";
+    private final static String UNSENT_FORM_TIME_KEY = "unsent-time-limit";
+
+    private final CommCareHomeActivity activity;
+    private SquareButtonWithNotification startButton;
+    private SquareButtonWithNotification logoutButton;
+    private SquareButtonWithNotification viewIncompleteFormsButton;
+    private SquareButtonWithNotification syncButton;
+
+    private HomeScreenAdapter adapter;
+    private GridViewWithHeaderAndFooter gridView;
+    private StaggeredGridView newGridView;
+    private View mTopBanner;
+
+    private String syncKey = "home.sync";
+    private String lastMessageKey = "home.sync.message.last";
+    private String homeMessageKey = "home.start";
+    private String logoutMessageKey = "home.logout";
+    private Pair<Long, int[]> syncDetails;
```

Comment:
this syncDetails value seems like the weirdest remaining thing here because it has a really weird class type and is the only non-UI statefulness in this class. Seems like the statefulness is unavoidable, maybe we should split out time since last sync/number unsent/number incomplete into separate variables? dunno